src/Libraries/RevitServices/Persistence/ElementIDLifecycleManager.cs

```diff
@@ -100,8 +100,13 @@ public int UnRegisterAssociation(T elementID, Object wrapper)
                 //ID already existed, check we're not over adding
                 if (existingWrappers.Contains(wrapper))
                 {
-                    int index = existingWrappers.FindIndex((x) => object.ReferenceEquals(x, wrapper));
-                    existingWrappers.RemoveAt(index);
+                    var removeList = existingWrappers.FindAll((x) => object.ReferenceEquals(x, wrapper));
```

Comment:
what case triggered this change? Could index ever be -1? (not found?)

Reply:
When I debug disable/enable transaction, I found that existingWrappers contains multiple wrapper, but no one can be referenceEqueals.

Reply:
This may be debugging code, I will solve this, but I think it would be better to add a judgment protection to the index.

Reply:
Thank you @ZiyunShang for the update.
I still have concern on this implementation. This requires Dynamo developer to keep in mind that the node may run in DisableTransaction mode, and need to deal with the behavior. And it also requires Dynamo developers to be familiar with the API internal implementation, know if the API calls regen or starts Transaction internally.
Moreover, even doing all above, this only works for the built-in nodes. It doesn't work for package nodes, python nodes, custom nodes, which are also in the AC list. 
@saintentropy @QilongTang any ideas?